app/models/User.rb

```diff
@@ -17,4 +18,17 @@ def password=(password)
     @password = password
     self.password_hash = BCrypt::Password.create(password)
   end
+
+  def self.authenticate(email,password)
+  user = first(email: email)
+    if user 
```

Comment:
I think this could be cleaned up to something like `user if user && BCrypt::Password.new(user.password_hash) == password`. I am not 100% sure what the 0 and -1 is for